src/auth/src/build_errors.rs

```diff
@@ -48,6 +48,11 @@ impl Error {
         matches!(self.0, ErrorKind::MissingField(_))
     }
 
+    /// The credential type is not supported for the given use case.
+    pub fn is_not_supported(&self) -> bool {
+        matches!(self.0, ErrorKind::NotSupported(_))
+    }
```

Comment:
Should this be protected by the configuration flag too?  Or are you certain this API will not change or get removed?

If we are ready to add this: this is a new feature, we should bump the minor version of the auth at the same time.